### ISSUE TITLE
Separate the svelte component and the js wrapper

### ISSUE THREAD

**Issue by AlexRMU** [josdejong/svelte-jsoneditor#372]
It would be nice to bring all the functionality that is needed for use without svelte into a separate component that will be imported from js.
And the pure svelte component will be imported from svelte.
You also need to make native events in the svelte component (and pass them above), and use callbacks in the wrapper, etc.

**Comment by josdejong** [comment]
I'm not entirely sure what you mean, do you mean to move the stuff related to the `vanilla-jsoneditor` into a separate repo? There is no "js wrapper", it is only a build script that generates a vanilla JS bundle using rollup.

**Comment by AlexRMU** [comment]
The main component has updateProps and other things that are required for vanilla js. This can be moved to a separate component. In addition, it has a size of 500 lines, it is better to reduce it somehow.
Plus, you can add a bunch of slots to be able to replace some components.

**Comment by AlexRMU** [comment]
Blocked by https://github.com/josdejong/svelte-jsoneditor/issues/373

**Comment by josdejong** [comment]
Yeah, that can be interesting. It would be neat to separate the stuff related to the `vanilla-package` from the svelte-version. It should not introduce friction though to implement a feature or fix and then deploy both packages (right now all of that goes in one go).

Is this something you would like to try out?
